#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_renew-gui.dpatch by Norbert Holze <norbert.holze@web.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes problems with libaqbanking 4.x

## Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnucash/+bug/426685
## Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=582048
## Patch: http://bugzilla-attachments.gnome.org/attachment.cgi?id=137440
## Description: fix crashes when transfering money online in Germany

@DPATCH@
diff -urNad gnucash-2.2.6~/src/import-export/aqbanking/gnc-ab-transfer.c gnucash-2.2.6/src/import-export/aqbanking/gnc-ab-transfer.c
--- gnucash-2.2.9/src/import-export/aqbanking/gnc-ab-transfer.c	2008-12-08 23:10:20.000000000 +0100
+++ gnucash-2.2.9-mod/src/import-export/aqbanking/gnc-ab-transfer.c	2009-06-26 21:07:42.022961995 +0200
@@ -236,6 +236,13 @@
             /* Create a context to store possible results */
             context = AB_ImExporterContext_new();
 
+            gui = gnc_GWEN_Gui_get(parent);
+            if (!gui) {
+                g_warning("gnc_ab_maketrans: Couldn't initialize Gwenhywfar GUI");
+                aborted = TRUE;
+                goto repeat;
+            }
+
             /* Finally, execute the job */
             AB_Banking_ExecuteJobs(api, job_list, context, 0);
 
